src/sentry/incidents/metric_issue_detector.py

```diff
@@ -319,6 +321,18 @@ def update(self, instance: Detector, validated_data: dict[str, Any]):
                     "Failed to send data to Seer, cannot update detector"
                 )
 
+        elif (
+            instance.config.get("detection_type") == AlertRuleDetectionType.DYNAMIC
+            and validated_data.get("config", {}).get("detection_type")
+            != AlertRuleDetectionType.DYNAMIC
+        ):
```

Comment:
for readability, it might help to make these variables; then it could read like: `if not is_instance_anomaly_detector and is_update_anomaly_detector`.